Add page about official OpenTofu Docker image (#1271)
opentofu/opentofu: website/docs/intro/install/docker.mdx

---
sidebar_position: 99

> sidebar_label: Docker Image

Reviewer: ```suggestion
sidebar_label: Container Image
```

description: |-

>   Use official Docker Image available on GitHub Container Registry.

Reviewer: ```suggestion
  Use official container image available on the GitHub Container Registry.
```

---

import CodeBlock from '@theme/CodeBlock';
import DockerPullScript from '!!raw-loader!./docker-pull.sh'
import DockerRunScript from '!!raw-loader!./docker-run.sh'

# Use OpenTofu as Docker Image

> OpenTofu is available as [Docker images](https://github.com/opentofu/opentofu/pkgs/container/opentofu),

Reviewer: ```suggestion
OpenTofu is available as [OCI container images](https://github.com/opentofu/opentofu/pkgs/container/opentofu),
```

and distributed via public GitHub Packages registry.

## Versions

Images are hosted as packages in opentofy GitHub organization. See the list
of available versions [here](https://github.com/opentofu/opentofu/pkgs/container/opentofu/versions?filters%5Bversion_type%5D=tagged).

These multi-arch tags are available:

- `latest`: latest overall version of OpenTofu,
- `Major`: a specific major version of OpenTofu,
- `Major`.`Minor`: a specific minor version of OpenTofu,
- `Major`.`Minor`.`Patch`: a specific patch version of OpenTofu.

To pull platform-specific images (`amd64`, `arm`, `arm64`, `386`) use:

- `<Version>`-`<Platform>`: a platform specific version of OpenTofu.

## Usage

> To be able to pull the image generate a PAT with scope `packages:read` and then
> use it to [login][ghcr-login] to GitHub Packages registry.

Reviewer: This is incorrect, you can pull without logging in, but rate limits may apply (see suggestion below).

```suggestion
```

To pull the image from GitHub Packages registry:

<CodeBlock language="bash">{DockerPullScript}</CodeBlock>

To run OpenTofu as a Docker container:

<CodeBlock language="bash">{DockerRunScript}</CodeBlock>

> [ghcr-login]: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry

Reviewer: ```suggestion
:::tip

If you run into rate limiting issues with GitHub, create a [Personal Access Token](https://github.com/settings/tokens) and [log in to the registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry).

:::
```

import DockerPullScript from '!!raw-loader!./docker-pull.sh'
import DockerRunScript from '!!raw-loader!./docker-run.sh'

# Use OpenTofu as Docker Image

OpenTofu is available as [OCI container images](https://github.com/opentofu/opentofu/pkgs/container/opentofu),
and distributed via public GitHub Packages registry.

## Versions

Images are hosted as packages in opentofy GitHub organization. See the list
of available versions [here](https://github.com/opentofu/opentofu/pkgs/container/opentofu/versions?filters%5Bversion_type%5D=tagged).

The multi platform images are available using the following tags:

- `latest`: latest overall version of OpenTofu,
- `Major`: a specific major version of OpenTofu,
- `Major`.`Minor`: a specific minor version of OpenTofu,
- `Major`.`Minor`.`Patch`: a specific patch version of OpenTofu.

To pull platform-specific images (`amd64`, `arm`, `arm64`, `386`) use:

- `<Version>`-`<Platform>`: a platform specific version of OpenTofu.

## Usage

To pull the image from GitHub Packages registry:

<CodeBlock language="bash">{DockerPullScript}</CodeBlock>

To run OpenTofu as a Docker container:

<CodeBlock language="bash">{DockerRunScript}</CodeBlock>

> :::tip
> If you run into rate limiting issues with GitHub, create a
> [Personal Access Token](https://github.com/settings/tokens) and
> [log in](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
> to the registry.
> :::

Reviewer: Just because this may cause issues with Docusaurus, two line breaks:
```suggestion
:::tip

If you run into rate limiting issues with GitHub, create a
[Personal Access Token](https://github.com/settings/tokens) and
[log in](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
to the registry.

:::
```
